cart.js

```diff
@@ -0,0 +1,78 @@
+/*
+IMPORTANT: some of this code contains more than what we have learned inside of class so far.
+I have marked the sections where you need to change something with //Task #: what you need to do
+*/
+
+//Task 1: retrieve user name from local storage, if it does assign to this variable
+let userName = localStorage.getItem("userName");
+let userH1 = document.getElementById("greet-user");
+userH1.textContent = userName;
+
+//Task 4: read from localStorage productsInCart array (make sure to parse)
+let productsInCart = JSON.parse(localStorage.getItem("productsInCarty"));
+
+//Note: this gets the ul for our products
+let productsUL = document.getElementById("cart-products");
+
+//Note: this adds our items in cart to the page, we will learn how to do this later
+function drawProducts(productsToDraw) {
+  //Note: clear out ul for redraw
+  productsUL.innerHTML = "";
+  //Note: add list item for each product in cart
+  /* 
+    This is what an example li will look like
+    <li>
+        <h3>Duct tape</h3>
+        <p><b>$4.72</b></p>
+        <img alt="duct tape" src="assets/duct-tape.png"/>
+        <button onclick="removeFromCart(0)">Remove from cart</button> 
+    </li>
+  */
+
+  let totalPrice = 0;
+  for (let index = 0; index < productsToDraw.length - 1; index++) {
```

Comment:
the other one @jvannier 
```suggestion
  for (let index = 0; index < productsToDraw.length; index++) {
```

Reply:
Oh right because < instead of <=